crates/rpc/src/server/accept.rs

```diff
@@ -162,61 +193,88 @@ impl<'a> TryFrom<&'a str> for AcceptHeaderValue<'a> {
     }
 }
 
-// ACCEPT TESTS
+// HEADER VERIFICATION TESTS
 // ================================================================================================
 
 #[cfg(test)]
 mod tests {
+    use http::{HeaderMap, HeaderName, HeaderValue};
+    use miden_objects::Digest;
     use semver::Version;
 
-    use super::{AcceptHeaderValue, AcceptLayer};
+    use super::HeaderVerificationLayer;
 
-    #[test]
-    fn current_version_is_parsed_and_matches() {
-        let a = AcceptLayer::new().unwrap();
-        let version = env!("CARGO_PKG_VERSION");
-        let version = Version::parse(version).unwrap();
-        assert!(a.version_req.matches(&version));
-    }
+    const TEST_GENESIS_COMMITMENT: &str =
+        "0x00000000000000000000000000000000000000000000000000000000deadbeef";
+    const TEST_RPC_VERSION: Version = Version::new(1, 2, 3);
 
-    #[test]
-    fn same_minor_different_patch_matches() {
-        let a = AcceptLayer::new().unwrap();
-        let version = env!("CARGO_PKG_VERSION");
-        let mut version = Version::parse(version).unwrap();
-        version.patch += 1;
-        assert!(a.version_req.matches(&version));
+    impl HeaderVerificationLayer {
+        fn for_tests() -> Self {
```

Comment:
can we `#[cfg(test)]` this fn?

Reply:
It already is because its in `mod tests`